internal/worker/server.go

```diff
@@ -174,6 +174,94 @@ func (s *Server) CheckBuildDependencies(dep uuid.UUID, jobErr *clienterrors.Erro
 	return nil
 }
 
+// DependencyChainErrors recursively gathers all errors from job's dependencies,
+// which caused it to fail. If the job didn't fail, `nil` is returned.
+func (s *Server) JobDependencyChainErrors(id uuid.UUID) (*clienterrors.Error, error) {
+	jobType, err := s.JobType(id)
+	if err != nil {
+		return nil, err
+	}
+
+	var jobResult *JobResult
+	var jobDeps []uuid.UUID
+	switch jobType {
+	case WorkerJobTypeOSBuild:
+		var osbuildJR OSBuildJobResult
+		_, jobDeps, err = s.OSBuildJobStatus(id, &osbuildJR)
```

Comment:
I wonder if we can make a generic `JobResult()` method that would just return deps and a job result. It should be technically possible, right?

Reply:
I actually had the same idea, but did it this way in the end. Yes, it should be possible to implement...

Reply:
I would propose at least exploring this idea. The big `switch` will need an update whenever a new job is added, whereas job results are meant to be fairly stable, right?

Reply:
I agree and plan to rework this part... 😉

Reply:
I looked at the code and remembered why I implemented the function this way...

Each job type specific status method checks the job result in a slightly different way and sets the `result.JobError` to a specific value.

E.g.
https://github.com/osbuild/osbuild-composer/blob/335c597452ab47b42dee9fbca56e2c1c01a54153/internal/worker/server.go#L178-L191
vs.
https://github.com/osbuild/osbuild-composer/blob/335c597452ab47b42dee9fbca56e2c1c01a54153/internal/worker/server.go#L263-L269

For this reason I incline to keeping the implementation as is. I'll add a note to the commit message explaining why I took this approach.

Reply:
Was the commit message updated?

Reply:
Obviously not, sorry... 🙄

Reply:
I amended the commit message.